diambra/arena/arena_gym.py

```diff
@@ -130,411 +170,176 @@ def render(self, mode='human', wait_key=1):
-
-            if k == "stage":
-                continue
-
-            if k[-2:] == "P1":
-                knew = "own" + k[:-2]
-            else:
-                knew = "opp" + k[:-2]
-
-            if v[0] == 0 or v[0] == 2:  # Discrete spaces
-                player_spec_dict[knew] = spaces.Discrete(v[2] + 1)
-            elif v[0] == 1:  # Box spaces
-                player_spec_dict[knew] = spaces.Box(low=v[1], high=v[2],
-                                                    shape=(1,), dtype=np.int32)
-
-            else:
-                raise RuntimeError("Only Discrete and Box Spaces allowed")
-
-        actions_dict = {
-            "move": spaces.Discrete(self.n_actions[0][0]),
-            "attack": spaces.Discrete(self.n_actions[0][1])
-        }
-
-        player_spec_dict["actions"] = spaces.Dict(actions_dict)
-        player_dict_p1 = player_spec_dict.copy()
-        observation_space_dict["P1"] = spaces.Dict(player_dict_p1)
-
-        actions_dict = {
-            "move": spaces.Discrete(self.n_actions[1][0]),
-            "attack": spaces.Discrete(self.n_actions[1][1])
-        }
-
-        player_spec_dict["actions"] = spaces.Dict(actions_dict)
-        player_dict_p2 = player_spec_dict.copy()
-        observation_space_dict["P2"] = spaces.Dict(player_dict_p2)
-
-        observation_space_dict["stage"] = spaces.Box(low=self.ram_states["stage"][1],
-                                                     high=self.ram_states["stage"][2],
-                                                     shape=(1,), dtype=np.int8)
-
-        self.observation_space = spaces.Dict(observation_space_dict)
-
-    def ram_states_integration(self, frame, data):
-
-        observation = {}
-        observation["frame"] = frame
-        observation["stage"] = np.array([data["stage"]], dtype=np.int8)
-
-        for ielem, elem in enumerate(["P1", "P2"]):
-
-            player_spec_dict = {}
-
-            # Adding env additional observations (side-specific)
-            for k, v in self.ram_states.items():
-
-                if k == "stage":
-                    continue
-
-                if k[-2:] == elem:
-                    knew = "own" + k[:-2]
-                else:
-                    knew = "opp" + k[:-2]
-
-                # Box spaces
-                if v[0] == 1:
-                    player_spec_dict[knew] = np.array([data[k]], dtype=np.int32)
-                else:  # Discrete spaces (binary / categorical)
-                    player_spec_dict[knew] = data[k]
-
-            actions_dict = {
-                "move": data["moveAction{}".format(elem)],
-                "attack": data["attackAction{}".format(elem)],
-            }
-
-            player_spec_dict["actions"] = actions_dict
-            observation[elem] = player_spec_dict
+    # Return the no-op action
+    def get_no_op_action(self):
+        no_op_action = {}
+        for idx in range(2):
+            if self.env_settings.action_space[idx] == "multi_discrete":
```

Comment:
Same, also consider turning the reassigning into a function you can use above as well

Reply:
See previous comment.